nni/algorithms/compression/v2/pytorch/pruning/tools/task_generator.py

```diff
@@ -43,13 +43,15 @@ def __init__(self, total_iteration: int, origin_model: Module, origin_config_lis
         keep_intermediate_result
             If keeping the intermediate result, including intermediate model and masks during each iteration.
         """
-        self.current_iteration = 0
-        self.target_sparsity = config_list_canonical(origin_model, origin_config_list)
         self.total_iteration = total_iteration
-
-        super().__init__(origin_model, origin_config_list=self.target_sparsity, origin_masks=origin_masks,
+        super().__init__(origin_model, origin_config_list=origin_config_list, origin_masks=origin_masks,
                          log_dir=log_dir, keep_intermediate_result=keep_intermediate_result)
 
+    def reset(self, model: Module, config_list: List[Dict] = [], masks: Dict[str, Dict[str, Tensor]] = {}):
```

Comment:
should we call `reset` in init?

Reply:
yes, `reset` is called in `super().__init__(...)`